### PR TITLE
635: Add census link for credited reviewers

### PR DESCRIPTION
**Pull request by lgxbslgx**
Hi all,

This patch uses the full name and the census link of the credited reviewer so that the `### Reviewers` part can be clearer.

For example, the [PR-6562](https://github.com/openjdk/jdk/pull/6562) has a credited reviewer which is shown as `jbhateja - Committer  Added manually`. 

After this patch, it will show: `Jatin Bhateja - Committer  Added manually`. The full name `Jatin Bhateja` will replace the username `jbhateja`.

If these is a census link, it will show: `[Jatin Bhateja](https://openjdk.java.net/census#jbhateja) - Committer  Added manually`.

And the test cases are added.

Thanks for taking the time to review.

Best Regards,
-- Guoxiong

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed

### Issue
 * [SKARA-635](https://bugs.openjdk.java.net/browse/SKARA-635): Add census link for credited reviewers

### Reviewers
 * [Erik Joelsson](https://openjdk.java.net/census#erikj) (@erikj79 - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/skara pull/1252/head:pull/1252` \
`$ git checkout pull/1252`

Update a local copy of the PR: \
`$ git checkout pull/1252` \
`$ git pull https://git.openjdk.java.net/skara pull/1252/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1252`

View PR using the GUI difftool: \
`$ git pr show -t 1252`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/skara/pull/1252.diff">https://git.openjdk.java.net/skara/pull/1252.diff</a>

</details>

### CODE REVIEW

File: bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java [hunk 1]
```diff
@@ -370,29 +371,60 @@ private String getRole(String username) {
...
+            ret.append(user.username());
             ret.append(" - ");
             ret.append(getRole(contributor.username()));
             ret.append(")");
             return ret.toString();
+        } else if (contributor == null) {
+            // The HostUser is not null and the Contributor is null
+            ret.append("@");
+            ret.append(user.username());
+            ret.append(" (Unknown ");
```
Comment (erikj79) [inline bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java:403]: I think I prefer the original message "no known <namespace> user name / role".

**Review by erikj79 — COMMENTED** [review]
Looks good otherwise.

File: bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java (hunk 1 above)
Comment (lgxbslgx) [reply bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java]: Fixed.

**Review by erikj79 — APPROVED** [review]

**Comment by lgxbslgx** [comment]
@erikj79 Thanks for the review. Could I get your help to sponsor this patch?

/integrate

**Pull request closed without merging**